src/app/shared/rss-feed/rss.component.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  OnDestroy,
+  OnInit,
+  ViewEncapsulation
+} from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { GroupDataService } from '../../core/eperson/group-data.service';
+import { LinkHeadService } from '../../core/services/link-head.service';
+import { ConfigurationDataService } from '../../core/data/configuration-data.service';
+import { getFirstCompletedRemoteData } from '../../core/shared/operators';
+import { environment } from '../../../../src/environments/environment';
+import { SearchConfigurationService } from '../../core/shared/search/search-configuration.service';
+import { SortOptions } from '../../core/cache/models/sort-options.model';
+import { PaginationService } from '../../core/pagination/pagination.service';
+import { Router } from '@angular/router';
+
+
+/**
+ * The Rss feed button componenet.
+ */
+@Component({
+  exportAs: 'rssComponent',
+  selector: 'ds-rss',
+  styleUrls: ['rss.component.scss'],
+  templateUrl: 'rss.component.html',
+  changeDetection: ChangeDetectionStrategy.Default,
+  encapsulation: ViewEncapsulation.Emulated
+})
+export class RSSComponent implements OnInit, OnDestroy  {
+
+  route$: BehaviorSubject<string>;
+
+  isEnabled$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+
+  uuid: string;
+  configuration$: Observable<string>;
+  sortOption$: Observable<SortOptions>;
+
+  constructor(private groupDataService: GroupDataService,
+              private linkHeadService: LinkHeadService,
+              private configurationService: ConfigurationDataService,
+              private searchConfigurationService: SearchConfigurationService,
+              private router: Router,
+              protected paginationService: PaginationService) {
+  }
+  ngOnDestroy(): void {
+    this.linkHeadService.removeTag("rel='alternate'");
+  }
+
+  ngOnInit(): void {
+    this.configuration$ = this.searchConfigurationService.getCurrentConfiguration('default');
+
+    this.configurationService.findByPropertyName('websvc.opensearch.enable').pipe(
+      getFirstCompletedRemoteData(),
+    ).subscribe((result) => {
+      const enabled = (result.payload.values[0] === 'true');
+      this.isEnabled$.next(enabled);
+    });
+
+    this.searchConfigurationService.getCurrentQuery('').subscribe((query) => {
+      this.sortOption$ = this.paginationService.getCurrentSort(this.searchConfigurationService.paginationID, null, true);
+      this.sortOption$.subscribe((sort) => {
+        this.uuid = this.groupDataService.getUUIDFromString(this.router.url);
+
+        const route = environment.rest.baseUrl + this.formulateRoute(this.uuid, sort, query);
+        this.addLinks(route);
+        this.linkHeadService.addTag({
+          href: environment.rest.baseUrl + '/opensearch/service',
```

Comment:
Just realized (after submitting my review):  I *think* we may need to avoid hardcoding this URL.  It appears, from the backend, this `/opensearch` path is configurable (via `websvc.opensearch.svccontext`). That means the UI code will break if you change it's configuration to something which is not "opensearch".

Likely the backend needs to make `websvc.opensearch.svccontext` accessible to the frontend (like `websvc.opensearch.enable` already is), so that the UI can read that value and build this URL appropriately.

Reply:
Correct I saw this configuration and added it to rest.cfg so we could pull it on the angular side in my most recent commit